Zend/zend_compile.c

```diff
@@ -5245,6 +5245,29 @@ void zend_compile_declare(zend_ast *ast) /* {{{ */
 				CG(active_op_array)->fn_flags |= ZEND_ACC_STRICT_TYPES;
 			}
 
+		} else if (zend_string_equals_literal_ci(name, "cache")) {
+			zval value_zv;
+
+			if (FAILURE == zend_declare_is_first_statement(ast)) {
+				zend_error_noreturn(E_COMPILE_ERROR, "cache declaration must be "
+					"the very first statement in the script");
```

Comment:
This message is a bit misleading, when cache=0 is used together (and after) with strict_types.